### PR TITLE
Schema manipulation

### LINKED ISSUE #25: PR: Add schema manipulation functionality
Should be able to manipulate the schema. Try altering schema using postico and see how it does this.

### LINKED ISSUE #29: PR: Uniform Create Table Syntax
## Summary
`SELECT sql FROM sqlite_master WHERE name='${table}';` returns the sqlite statement used to create `table`.  However, the format/syntax isn't the same for all instances as it is determined by the user/client and sqlite allows multiply different syntaxes for creating tables/columns/constraints. 

There should be an adapter/function that takes any sqlite create table statement and returns a consistent format that falcon-core can work with

### Example
Editing with db-browser will force an sqlite table's creation statement to look like this, regardless of initial format/syntax

image

### PR DESCRIPTION
**Pull request by jooohhn**
**Summary**: Added ability for adjusting a table's schema/structure

- Methods added
   - [x] `renameTable`
   - [x] `dropTable`
   - [x] `addTableColumn`
   - [x] `renameTableColumn`
   - [x] `dropTableColumns`
- Helper Methods: 
   - `getCreateTableSql(table)`: Gets the sqlite statement that creates a `table`. Should do so in a uniform and consistent way regardless of how the client created the table
   - `getTablePropertiesSql(table)`: Gets the properties that SQLite's table creation requires. Should do so in a uniform and consistent way regardless of how the client created the table
       - i.e. The statements (arguments) within the parentheses of a table creation
       - e.g. Columns and their properties, primary key definition, foreign key definitions, etc.
- Issues Closed
   - Closes #25  
   - Closes #29

### CODE REVIEW

File: src/db/provider_clients/SqliteProviderFactory.js [hunk 1]
```diff
@@ -123,20 +125,21 @@ class SqliteProvider extends BaseProvider implements ProviderInterface {
...
       return `(${rowData.join(', ')})`;
     });
     const query = `
-     INSERT INTO ${table} (${tableKeys.join(', ')})
+     INSERT INTO ${table} (${tableColumns.join(', ')})
      VALUES
      ${rowSqls.join(',\n')};
     `;
-    return this.driverExecuteQuery({ query }).then(res => res.data);
+    const foo = this.driverExecuteQuery({ query }).then(res => res.data);
```
Comment (amilajack) [inline src/db/provider_clients/SqliteProviderFactory.js]: We can replace this line with just `return this.driverExecuteQuery({ query }).then(res => res.data);`

File: src/db/provider_clients/SqliteProviderFactory.js [hunk 2]
```diff
@@ -456,6 +626,14 @@ class SqliteProvider extends BaseProvider implements ProviderInterface {
             return reject(err);
           }
 
+          // const profile = new Promise(_resolve =>
```
Comment (amilajack) [inline src/db/provider_clients/SqliteProviderFactory.js]: We should remove this commented code

File: src/db/provider_clients/ProviderInterface.js [hunk 3]
```diff
@@ -150,12 +150,15 @@ export interface ProviderInterface {
...
   insert: (
     table: string,
     values: Array<{ [value: string]: any }>
-  ) => Promise<bool>,
-  update: (table: string, records: Array<Object>) => Promise<bool>,
+  ) => Promise<{ timing: number }>,
+  update: (
+    table: string,
+    records: Array<Object>
+  ) => Promise<{ timing: number }>,
```
Comment (amilajack) [inline src/db/provider_clients/ProviderInterface.js:162]: `renameTableColumn()` is missing from the interface. Also shouldnt this api be renamed to `renameTableColumns()`?

**Review by amilajack — CHANGES_REQUESTED** [review]
Let's add all these new methods to the `ProviderInterface`

File: src/db/provider_clients/ProviderInterface.js (hunk 3 above)
Comment (jooohhn) [reply src/db/provider_clients/ProviderInterface.js:162]: I'll add the methods to the interface. Also `update` is for updating a record in a table, which is different from modifying table schema/structure

**Pull request merged**